### PR TITLE
Fixes #10601: If component value starts with # , report is not outputed by rudder agent output

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/10601

### CODE REVIEW

File: share/lib/reports.awk [hunk 1]
```diff
@@ -113,28 +111,33 @@ function print_report_multiline() {
...
+    # 1 is R:
     technique = r[2];
     state = r[3];
+    # 4 is rule ID
+    directiveid = r[5];
+    # 6 is generation
     component = r[7];
     key = r[8];
-    directiveid = r[5];
+    rest = r[9];
```
Comment (ncharles) [inline share/lib/reports.awk]: what if there is a @@ in the message ?

File: share/lib/reports.awk [hunk 2]
```diff
@@ -113,28 +111,33 @@ function print_report_multiline() {
...
+
+    if (match(rest, /##/)) { # match the first ##
+      # date then rest
+      rest = substr(rest,RSTART+RLENGTH)
+      if (match(rest, /@#/)) { # match the first @#
+        # node id then message
+        message = substr(rest,RSTART+RLENGTH)
```
Comment (ncharles) [inline share/lib/reports.awk]: we should check for r[10] and more, and if they exists, add them there

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
ping @amousset

**Comment by amousset** [comment]
It does not seem to work with 4.1, I'm investigating.

**Comment by amousset** [comment]
(with Rudder 4.1.3 on Ubuntu 16.04)

File: share/lib/reports.awk [hunk 3]
```diff
@@ -113,28 +111,38 @@ function print_report_multiline() {
...
     state = r[3];
+    # 4 is rule ID
+    directiveid = r[5];
+    # 6 is generation
     component = r[7];
     key = r[8];
-    directiveid = r[5];
+    # take the rest
+    rest = "";
+    for(i=9;i<n;i++) {
```
Comment (amousset) [inline share/lib/reports.awk]: `i<=n`

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**